code/Player/Controllers/HumanController.cs

```diff
@@ -147,6 +150,41 @@ private void Walk()
 			}
 			
 			Accelerate();
+
+			if (WishVelocity.IsNearZeroLength) return;
+			
+			ClimbStep();
+		}
+
+		/// <summary>
+		/// Tries to climb steps in case player can't just simply walk.
+		/// </summary>
+		private void ClimbStep() {
+			MoveHelper mover = CheckMove();
+			if (mover.TryMove(Time.Delta) > 0.9) return;
+			
+			TraceResult ray = Trace.Ray(mover.Position, mover.Position + Vector3.Up * StepSize)
+				.HitLayer(CollisionLayer.All, false)
+				.HitLayer(CollisionLayer.Solid, true)
+				.HitLayer(CollisionLayer.GRATE, true)
+				.HitLayer(CollisionLayer.PLAYER_CLIP, true)
+				.Ignore(Pawn)
+				.Run();
+
+			Position = ray.EndPos;
+		}
+
+		/// <summary>
+		/// Try to do simple walk
+		/// </summary>
+		/// <returns>Returns if simple walk was possible</returns>
```

Comment:
It returns a MoveHelper, not an "if simple walk was possible" 
Basicaly the user of this method can't really understind what to do with the return